### ISSUE TITLE
same issue here, the detail of how to use item side information to get the "dataset"_rel.npy is unknown, and the paper description of this part is sparse. (e.g. for the Amazon dataset which item attribute is used and how to determine whether items are bidirectional or unidirectional links)

### ISSUE THREAD

**Issue by zhihaowang9** [zhuty16/GES#4]
same issue here, the detail of how to use item side information to get the "dataset"_rel.npy is unknown, and the paper description of this part is sparse. (e.g. for the Amazon dataset which item attribute is used and how to determine whether items are bidirectional or unidirectional links)

_Originally posted by @Li-fAngyU in https://github.com/zhuty16/GES/issues/1#issuecomment-1579757313_
Same question about the "dataset"_rel.npy .For yelp and goole dataset, the max and min values are both 0, for amazon the max value is 3 and min value is 0. But in the paper the value should be 0, 1, 2.

**Comment by zhihaowang9** [comment]
> ```
>           same issue here, the detail of how to use item side information to get the "dataset"_rel.npy is unknown, and the paper description of this part is sparse. (e.g. for the Amazon dataset which item attribute is used and how to determine whether items are bidirectional or unidirectional links)
> ```
> 
> _Originally posted by @Li-fAngyU in [#1 (comment)](https://github.com/zhuty16/GES/issues/1#issuecomment-1579757313)_ Same question about the "dataset"_rel.npy .For yelp and goole dataset, the max and min values are both 0, for amazon the max value is 3 and min value is 0. But in the paper the value should be 0, 1, 2.

In the section 5.1, I found the setting of these 3 datasets, but what about the processing of new dataset? The github code doesn't show the preprocessing step.

**Comment by zhuty16** [comment]
Thank you for your attention. In Section 5.1 of our paper, we provided details on how we constructed the item semantic relations for the three datasets. For example, for the Amazon dataset, we utilized metadata available at https://jmcauley.ucsd.edu/data/amazon/. Our approach constructs connections between items based on their presence in the "related" lists, namely "also bought", "also viewed", "bought after viewing", and "bought together". Regarding new datasets, you have the flexibility to construct item semantic relations based on the specific characteristics and scenarios of your data. The effectiveness of item semantic relations in improving recommendations may depend on the quality of these relations.
